STEP/CodeGeneration/StandardEnvironmentVisitor.cs

```diff
@@ -0,0 +1,307 @@
+﻿using STEP.AST;
+using STEP.AST.Nodes;
+
+namespace STEP.CodeGeneration;
+
+/// <summary>
+/// A visitor that replaces the names of any standard functions with their corresponding Arduino names.
+/// </summary>
+public class StandardEnvironmentVisitor : IVisitor
+{
+    public void Visit(AndNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(OrNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(EqNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(NeqNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(GThanNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(GThanEqNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(LThanNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(LThanEqNode n)
+    {
+        n.Left.Accept(this);
+        n.Right.Accept(this);
+    }
+
+    public void Visit(NegNode n)
+    {
+        n.Left.Accept(this);
+    }
+
+    public void Visit(NumberNode n)
+    {
+
+    }
+
+    public void Visit(StringNode n)
+    {
+        
+    }
+
+    public void Visit(BoolNode n)
+    {
+        
+    }
+
+    public void Visit(ArrDclNode n)
+    {
+        n.Left.Accept(this);
+        n.Right?.Accept(this);
+        n.IdRight?.Accept(this);
```

Comment:
```suggestion
        n.Left.Accept(this);
        n.Right?.Accept(this);
```
IdRight findes ikke længere, så du vil komme i problemer hvis det bliver merged ind i dev!

Reply:
det er fexed!